cbmc_viewer/coveraget.py

```diff
@@ -227,6 +234,46 @@ def merge_coverage_data(coverage_list):
         raise UserWarning("Error merging coverage data: {}".format(error)) from error
     return coverage
 
+def filter_reachable_functions(coverage, reachable_functions=None):
+    """Restrict coverage data to coverage for reachable functions"""
+
+    if not reachable_functions:
+        return coverage
+
+    def valid_coverage_data(coverage, reachable):
+        valid = True
+        for file_name, file_coverage in coverage.items():
+            for func_name, func_coverage in file_coverage.items():
+                if func_name in reachable.get(file_name, {}):
+                    if not func_coverage:
+                        logging.warning("Reachable function %s has no coverage data", func_name)
+                        valid = False
+                else:
+                    lines = [int(line) for line, status in func_coverage.items()
+                             if status is not Status.MISSED]
+                    if lines:
+                        logging.warning(
+                            "Unreachable function %s has reachable lines %s",
+                            func_name, ', '.join([str(line) for line in sorted(lines)])
+                        )
```

Comment:
How can we have a unreachable function with reachable lines?

Reply:
I'm concerned about the diffblue tools providing inconsistent information, I tried to drive that home with comments in the code referring to sanity checks.